### ISSUE TITLE
Using custom rss user agents

### ISSUE THREAD

**Issue by benhartwich** [podlove/podlove-publisher#1199]
Hi,

for now podlove is using the OPAWG rss user agent list and the matomo user agent list. Is there a possility to add custom rss user agents, which podlove can recognize? E.g. you have got a bigger podcasting solution with podlove and there are custom podcast apps, you´re providing.

I think the best solution for getting to know how many are listening to your podcast at your own apps is to get a custom rss user agents recognized by podlove. Is this anyhow possible?

**Comment by eteubert** [comment]
I added a hook for this in the beta, it's actually just a single line of code so feel free to add it yourself until the next release: https://github.com/podlove/podlove-publisher/commit/0d9df3db92a3f555f6695e40b699005c0a430665

This allows you to add your own data in OPAWG structure, for example:

```php
add_filter('podlove_useragent_opawg_data', function ($data) {
    $data[] = (object) [
        'user_agents' => ['^Podcat 2'],
        'app' => 'Podcat',
        'os' => 'ios',
        'device' => 'phone'
    ];

    return $data;
});
```

⚠️ Note that the each entry needs to be casted to `(object)` as in the example.

The only required attributes are `user_agents` and `app`. See https://github.com/opawg/user-agents/tree/master/src for examples (in JSON format).

**Comment by eteubert** [comment]
For reference, this was implemented in 0d9df3db (just not release yet)
